filament/src/materials/dof.mat

```diff
@@ -49,17 +44,271 @@ fragment {
 
 #include "dofUtils.fs"
 
+
+#define RING_DENSITY    6.0
+#define RING_COUNT	    4.0
+#define SAMPLE_COUNT    ((RING_DENSITY * RING_COUNT * (RING_COUNT - 1.0) + 2.0) * 0.5)
+
+void dummy(){}
+
+layout(location = 1) out float outAlpha;
+
+
+float sampleWeight(float coc) {
+    // The contribution of sample is inversely proportional to *it's* area
+    // (the larger area, the fainter it is).
+    // In theory this factor should be 1 / pi * radius^2, however 1/pi is a constant, and
+    // because we divide by the total weight eventually, it can be ommited, in fact, any
+    // constant can be used. Here we use MAX_COC_RADIUS * MAX_COC_RADIUS instead of 1/pi,
+    // because this allows us to fit in mediump precision.
+    return (MAX_COC_RADIUS * MAX_COC_RADIUS) / (max(coc * coc, 0.5));
+}
+
+float intersection(float border, float absCoc, float mip) {
+    return saturate((absCoc - border) * pow(0.5, mip) + 0.5);
+}
+
+float rcp(float x) {
+    return 1.0 / x;
+}
+
+/*
+ * We use the accumulation method described in
+ * "Bokeh depth of field in a single pass" by Dennis Gustafsson
+ * [http://blog.tuxedolabs.com/2018/05/04/bokeh-depth-of-field-in-single-pass.html]
+ *
+ * We initialize the accumulator with the center value and our initial total weight starts at 1.0.
+ * Then, each time a new sample does not intersect with the center (meaning it doesn't contribute
+ * to this fragment), we scale the current accumulation to a weight of 'i + 1'.
+ * Samples that do contribute are simply acumulated, also increasing the total weight by 1.
+ * At the end of the process we're guaranteed to have a total weight of SAMPLE_COUNT + 1.
+ * The scale factor at each "miss" step is simply (i + 1) / i.
+ * With this technique the weight of each sample is constant; in reality it should depend on
+ * the circle-of-confusion area.
+ */
+
+void accumulateBackground(inout vec4 background, inout float opacity,
+        inout float i, highp vec2 pos, float radius, float mip) {
+    float coc = textureLod(materialParams_cocFgBg, pos, mip).g;
+    vec4 s = textureLod(materialParams_background, pos, mip);
+    float o = cocToAlpha(coc);
+    float w = intersection(radius, abs(coc), mip) * isBackground(coc);
+    background += mix(background * rcp(i), s, w);
+    opacity += mix(opacity * rcp(i), o, w);
+    i += 1.0;
+    //w *= sampleWeight(coc);
```

Comment:
Remove commented code

Reply:
Yeah, so I don't want to remove that one just yet, as we might switch to it soon, if I figure out some issues.